### DISCUSSION TITLE
Don't set empty CPU limits to CPU requests

### DISCUSSION THREAD

**Discussion by flixr** [flyteorg/flyte#3602] [RFC Incubator]
Currently a (container) task with only CPU requests but no limits set, still gets limits applied:

```
from flytekit import ContainerTask, kwtypes, workflow, Resources

hello_task = ContainerTask(
    name="hello",
    image="ubuntu:20.04",
    requests=Resources(cpu="2", mem="1Gi"),
    limits=Resources(mem="2Gi"),
    command=["echo", "hello"]
)
```
Running this task results in a pod with cpu limit also set to 2 (same as request), but there should be no cpu limit.

In many cases setting CPU limits (not requests) results in unused CPU due to throttling, see also 
https://home.robusta.dev/blog/stop-using-cpu-limits.

So I propose to only copy the requests to limits as a whole if limits are completely unset instead of filling missing limits with the respective requests values.

Originally I created this as a bug: https://github.com/flyteorg/flyte/issues/3574

**Comment by cosmicBboy** [comment]
hi @flixr this is a bug we want to address, but it doesn't fit the RFC criterion of being a significant change, gonna close this discussion and please feel free to contribute a PR, or ping the flyte snack [contribute](https://flyte-org.slack.com/archives/C04NJPLRWUX) channel if you want to raise awareness of this issue to the community

**Comment by cosmicBboy** [comment]
Closing this discussion, see #3574
